pkg/target/target.go

```diff
@@ -451,34 +489,95 @@ func (h *K8sValidationTarget) ToMatcher(u *unstructured.Unstructured) (constrain
 		if err != nil {
 			return nil, fmt.Errorf("%w: %v", ErrCreatingMather, err)
 		}
-		return &Matcher{match}, nil
+		if h.cache == nil {
```

Comment:
Another reason for this to not be a pointer - as-is has concurrency issues if `ToMatcher` and `ProcessData` are called concurrently. You'd need to add mutexes to guard read/write access to this pointer just so you can lazily initialize it, which doesn't seem worth it. (Just initialize nsCache.cache in `nsCache.Add`)